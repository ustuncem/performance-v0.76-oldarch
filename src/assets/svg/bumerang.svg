<svg width="351" height="390" viewBox="0 0 351 390" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M0.637288 10.4461C2.47004 4.67538 12.5539 0.276773 50.8091 0.857873C98.8107 1.58425 130.645 9.22735 185.963 21.3578C234.004 31.8904 269.947 39.267 311.065 61.4296C321.796 67.2084 332.623 73.8588 340.634 86.4009C355.783 110.105 350.168 138.087 343.544 168.837C337.446 197.125 327.772 219.167 308.677 262.095C301.941 277.228 289.912 303.475 271.519 335.476C251.221 370.778 239 391.65 229.188 389.574C215.709 386.717 214.692 342.222 214.627 338.227C213.691 282.013 241.265 266.56 243.29 210.765C244.008 190.862 244.921 165.721 230.206 142.033C217.709 121.921 200.332 113.148 136.138 86.8851C117.398 79.2178 98.6182 71.6796 79.7806 64.2545C70.8969 60.7598 61.9982 57.3216 53.1094 53.8432C40.1527 48.7746 28.6886 43.0361 17.6921 34.4568C11.7388 29.8242 -2.30561 19.6871 0.637288 10.4461Z" fill="#85D460"/>
</svg>
