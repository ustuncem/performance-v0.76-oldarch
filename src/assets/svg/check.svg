<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.5364 11.73C7.18492 11.3785 6.61508 11.3785 6.2636 11.73C5.91213 12.0814 5.91213 12.6513 6.2636 13.0027L9.53043 16.2696C9.8819 16.621 10.4517 16.621 10.8032 16.2696L18.5364 8.5364C18.8879 8.18492 18.8879 7.61508 18.5364 7.2636C18.1849 6.91213 17.6151 6.91213 17.2636 7.2636L10.1668 14.3604L7.5364 11.73Z" fill="#fill"/>
</svg>
