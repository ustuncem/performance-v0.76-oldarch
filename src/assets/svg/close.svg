<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M0 12C0 5.37258 5.37258 0 12 0C18.6274 0 24 5.37258 24 12C24 18.6274 18.6274 24 12 24C5.37258 24 0 18.6274 0 12Z" fill="#E5E5E5"/>
<path d="M9.16992 14.83L14.8299 9.16998" stroke="#262626" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M14.8299 14.83L9.16992 9.16998" stroke="#262626" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
