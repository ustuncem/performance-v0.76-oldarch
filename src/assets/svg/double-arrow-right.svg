<svg width="20" height="14" viewBox="0 0 20 14" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M5.67529 11.4333L9.32496 7.78366C9.75598 7.35264 9.75598 6.64733 9.32496 6.21632L5.67529 2.56665" stroke="#stroke" stroke-width="#stroke-width" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M11.5527 11.4333L15.2024 7.78366C15.6334 7.35264 15.6334 6.64733 15.2024 6.21632L11.5527 2.56665" stroke="#stroke" stroke-width="#stroke-width" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
