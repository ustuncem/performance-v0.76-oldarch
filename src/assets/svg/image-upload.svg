<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M11.9999 13.3333C13.4727 13.3333 14.6666 12.1394 14.6666 10.6667C14.6666 9.19391 13.4727 8 11.9999 8C10.5272 8 9.33325 9.19391 9.33325 10.6667C9.33325 12.1394 10.5272 13.3333 11.9999 13.3333Z" stroke="#stroke" stroke-width="#stroke-width" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M17.3334 2.66669H12.0001C5.33341 2.66669 2.66675 5.33335 2.66675 12V20C2.66675 26.6667 5.33341 29.3334 12.0001 29.3334H20.0001C26.6667 29.3334 29.3334 26.6667 29.3334 20V13.3334" stroke="#stroke" stroke-width="#stroke-width" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M21 6.66669H28.3333" stroke="#stroke" stroke-width="#stroke-width" stroke-linecap="round"/>
<path d="M24.6667 10.3333V3" stroke="#stroke" stroke-width="#stroke-width" stroke-linecap="round"/>
<path d="M3.56006 25.2666L10.1334 20.8533C11.1867 20.1466 12.7067 20.2266 13.6534 21.04L14.0934 21.4266C15.1334 22.32 16.8134 22.32 17.8534 21.4266L23.4001 16.6666C24.4401 15.7733 26.1201 15.7733 27.1601 16.6666L29.3334 18.5333" stroke="#stroke" stroke-width="#stroke-width" stroke-linecap="round" stroke-linejoin="round"/>
</svg>