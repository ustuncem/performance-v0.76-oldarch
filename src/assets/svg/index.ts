export const DoubleArrowRight = require('./double-arrow-right.svg');
export const Bumerang = require('./bumerang.svg');
export const ImageUpload = require('./image-upload.svg');
export const ArrowIcon = require('./arrow-icon.svg');
export const DownArrowIcon = require('./down-arrow-icon.svg');
export const Check = require('./check.svg');
export const Cross = require('./cross.svg');
export const Close = require('./close.svg');
