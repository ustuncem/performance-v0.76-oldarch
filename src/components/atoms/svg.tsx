import React, { forwardRef } from 'react';
import { View, ViewProps } from 'react-native';

import { SvgProps as RNSvgProps } from 'react-native-svg';

export interface SvgComponent {
  default: ({ ...props }: RNSvgProps) => JSX.Element;
}

export interface SvgProps extends ViewProps {
  Icon: SvgComponent;
  stroke?: string;
  strokeWidth?: number;
  fill?: string;
  width?: number | string;
  height?: number | string;
}

/**
 * Component that renders an SVG icon.
 * @param {SvgProps} props - The component props.
 * @param {SvgComponent} Icon - The SVG component to render.
 * @param {string} [stroke] - The stroke color for the SVG.
 * @param {string} [fill] - The fill color for the SVG.
 * @param {number|string} [width='100%'] - The width of the SVG. Defaults to '100%'.
 * @param {number|string} [height='100%'] - The height of the SVG. Defaults to '100%'.
 * @param {ViewProps} rest - Additional View props to be passed to the container.
 * @returns {React.ReactElement} The rendered SVG component wrapped in a View.
 */
const Svg = forwardRef<View, SvgProps>(
  (
    { Icon, stroke, strokeWidth, fill = 'transparent', width = '100%', height = '100%', ...props },
    ref
  ) => {
    const RenderedIcon = Icon?.default || Icon;

    return (
      <View {...props} ref={ref}>
        <RenderedIcon
          width={width}
          height={height}
          stroke={stroke}
          strokeWidth={strokeWidth}
          fill={fill}
        />
      </View>
    );
  }
);

Svg.displayName = 'Svg';

export default Svg;
