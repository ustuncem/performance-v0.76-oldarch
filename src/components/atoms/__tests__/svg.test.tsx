import React from 'react';

import { render } from '@testing-library/react-native';

import Svg from '#/components/atoms/svg';
import { ChevronRight as MockIcon } from '#assets/svg';

describe('Svg Component', () => {
  it('should render the Svg component with default props', () => {
    const { getByTestId } = render(<Svg Icon={MockIcon} testID="svg-container" />);
    const svgContainer = getByTestId('svg-container');

    expect(svgContainer).toBeTruthy();
  });

  it('should pass stroke and fill props to the Svg Icon component', () => {
    const stroke = '#ff0000';
    const fill = '#00ff00';

    const { getByTestId } = render(
      <Svg Icon={MockIcon} stroke={stroke} fill={fill} testID="svg-container" />
    );

    const svgIcon = getByTestId('svg-container').findByType(MockIcon);

    expect(svgIcon.props.stroke).toBe(stroke);
    expect(svgIcon.props.fill).toBe(fill);
  });

  it('should apply additional View props', () => {
    const style = { backgroundColor: 'blue' };

    const { getByTestId } = render(<Svg Icon={MockIcon} style={style} testID="svg-container" />);

    const svgContainer = getByTestId('svg-container');
    expect(svgContainer.props.style).toMatchObject(style);
  });

  it('should match the snapshot', () => {
    const stroke = '#ff0000';
    const fill = '#00ff00';

    const { toJSON } = render(<Svg Icon={MockIcon} stroke={stroke} fill={fill} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
