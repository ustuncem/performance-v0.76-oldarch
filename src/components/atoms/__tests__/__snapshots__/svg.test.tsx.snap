// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg Component should match the snapshot 1`] = `
<View>
  <SvgMock
    fill="#00ff00"
    height="100%"
    stroke="#ff0000"
    width="100%"
  />
</View>
`;
